core/src/test/java/com/microsoft/applicationinsights/internal/statsbeat/CustomDimensionsTest.java

```diff
@@ -0,0 +1,58 @@
+package com.microsoft.applicationinsights.internal.statsbeat;
+
+import com.microsoft.applicationinsights.internal.system.SystemInformation;
+import com.microsoft.applicationinsights.internal.util.PropertyHelper;
+import org.junit.Before;
+import org.junit.Test;
+
+import static com.microsoft.applicationinsights.internal.statsbeat.Constants.CUSTOM_DIMENSIONS_CIKEY;
+import static com.microsoft.applicationinsights.internal.statsbeat.Constants.CUSTOM_DIMENSIONS_OS;
+import static com.microsoft.applicationinsights.internal.statsbeat.Constants.CUSTOM_DIMENSIONS_RP;
+import static com.microsoft.applicationinsights.internal.statsbeat.Constants.CUSTOM_DIMENSIONS_RUNTIME_VERSION;
+import static com.microsoft.applicationinsights.internal.statsbeat.Constants.CUSTOM_DIMENSIONS_VERSION;
+import static com.microsoft.applicationinsights.internal.statsbeat.Constants.OS_LINUX;
+import static com.microsoft.applicationinsights.internal.statsbeat.Constants.OS_WINDOWS;
+import static com.microsoft.applicationinsights.internal.statsbeat.Constants.UNKNOWN;
+import static org.junit.Assert.assertEquals;
+
+public class CustomDimensionsTest {
+
+    @Before
+    public void setup() {
+        CustomDimensions.reset();
+    }
+
+    @Test
+    public void testResourceProvider() {
+        assertEquals(CustomDimensions.getInstance().getProperties().get(CUSTOM_DIMENSIONS_RP), UNKNOWN);
```

Comment:
the assertions are reversed in this test class, confusingly assertEquals takes expected as first arg